apps/docs/scripts/submit-newsletter-signup.ts

```diff
@@ -1,21 +1,36 @@
 'use server'
 
-export async function submitNewsletterSignup(email: string) {
-	const res = await fetch('https://api.sendgrid.com/v3/marketing/contacts', {
-		method: 'PUT',
-		headers: {
-			Authorization: `Bearer ${process.env.SENDGRID_API_KEY}`,
-			'Content-Type': 'application/json',
-		},
-		body: JSON.stringify({
-			list_ids: [process.env.SENDGRID_LIST_ID],
-			contacts: [
-				{
-					email,
+export async function submitNewsletterSignup(
+	email: string,
+	cookie: string | undefined,
+	url: string
+) {
+	const res = await fetch(
+		'https://api.hsforms.com/submissions/v3/integration/secure/submit/145620695/ff47937e-8c83-4f26-b00a-1ea4ae11a7c4',
+		{
+			method: 'POST',
+			headers: {
+				Authorization: `Bearer ${process.env.HUBSPOT_API_KEY}`,
+				'Content-Type': 'application/json',
+			},
+			body: JSON.stringify({
+				submittedAt: Date.now(),
+				fields: [
+					{
+						name: 'email',
+						value: email,
+					},
+				],
+				context: {
+					hutk: cookie,
```

Comment:
up above it's `hubspotutk` - does the context matter here that it's the same name or is `hutk` just fine?

Reply:
Feels like there are some inconsistencies on hubspots side:
> **hutk**
Body param
Used in context. The tracking cookie token value used for HubSpot lead activity tracking. You can retrieve this value from the "**hubspotutk**" cookie placed in the user's browser by the HubSpot JavaScript Tracking Code; the tracking code must be installed on the page that contains the form.
Please Note: While the **hutk** value is not required for the submission to be accepted, this token is used to associate analytics data with a contact record, so without this you will not see any page views or analytics source data for the contact record.